optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
+                    TestdataSolution solution = scoreDirector.getWorkingSolution();
+                    TestdataEntity entity = solution.getEntityList().get(x);
+                    scoreDirector.beforeVariableChanged(entity, "value");
+                    entity.setValue(solution.getValueList().get(x));
+                    scoreDirector.afterVariableChanged(entity, "value");
+                    scoreDirector.triggerVariableListeners();
+                });
+    }
+
+    private void assertDifferentSolveMethods(int problemCount, SolverManager<TestdataSolution, Integer> solverManager) throws InterruptedException, ExecutionException {
+        assertSolveWithoutConsumer(problemCount, solverManager);
+        assertSolveWithConsumer(problemCount, solverManager, true);
+        assertSolveWithConsumer(problemCount, solverManager, false);
+    }
+
+    private void assertSolveWithoutConsumer(int problemCount, SolverManager<TestdataSolution, Integer> solverManager)
+            throws InterruptedException, ExecutionException {
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>(problemCount);
+
+        for (int id = 0; id < problemCount; id++) {
+            jobs.add(solverManager.solve(id, PlannerTestUtils.generateTestdataSolution(Integer.toString(id))));
+        }
+        assertInitializedJobs(jobs);
+    }
+
+    private void assertSolveWithConsumer(
+            int problemCount, SolverManager<TestdataSolution, Integer> solverManager, boolean onlyFinalBestSolutions)
+            throws ExecutionException, InterruptedException {
+
+        // Two solutions for every problem.
+        Map<Integer, List<TestdataSolution>> consumedSolutions = new HashMap<>(problemCount * 2);
+
+        List<SolverJob<TestdataSolution, Integer>> jobs = new ArrayList<>(problemCount);
+
+        for (int id = 0; id < problemCount; id++) {
+            List<TestdataSolution> consumedBestSolutions = Collections.synchronizedList(new ArrayList<>());
+            String solutionName = Integer.toString(id);
+            if (onlyFinalBestSolutions) {
+                jobs.add(solverManager.solve(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            } else {
+                jobs.add(solverManager.solveAndListen(
+                        id,
+                        problemId -> PlannerTestUtils.generateTestdataSolution(solutionName, 2),
+                        consumedBestSolutions::add, null));
+            }
+            consumedSolutions.put(id, consumedBestSolutions);
+        }
+        assertInitializedJobs(jobs);
+        assertConsumedSolutions(problemCount, onlyFinalBestSolutions, consumedSolutions);
+    }
+
+    private void assertConsumedSolutions(int problemCount, boolean onlyFinalBestSolutions, Map<Integer, List<TestdataSolution>> consumedSolutions) {
```

Comment:
Can we rename "onlyFinalBestSolutions" to "listenWhileSolving" so it uses the same terminology as solveAndListen() vs solve(), to indicate that it's basically making that difference?